presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -307,28 +308,29 @@ protected Boolean visitFunctionCall(FunctionCall node, Void context)
         {
             if (metadata.isAggregationFunction(node.getName())) {
                 if (!node.getWindow().isPresent()) {
-                    AggregateExtractor aggregateExtractor = new AggregateExtractor(metadata.getFunctionRegistry());
-                    WindowFunctionExtractor windowExtractor = new WindowFunctionExtractor();
-
-                    for (Expression argument : node.getArguments()) {
-                        aggregateExtractor.process(argument, null);
-                        windowExtractor.process(argument, null);
-                    }
-
-                    if (!aggregateExtractor.getAggregates().isEmpty()) {
+                    List<FunctionCall> aggregateFunctions = ExpressionTreeUtils.extractExpressionsOfTypeUsingPredicate(
+                            node.getArguments(),
+                            FunctionCall.class,
+                            isAggregationPredicate(metadata.getFunctionRegistry()));
+                    List<FunctionCall> windowFunctions = ExpressionTreeUtils.extractExpressionsOfTypeUsingPredicate(
```

Comment:
Done, both on the static import and abstracting the logic into a separate method.